### DISCUSSION TITLE
When upgrading the jaeger operator version from 1.19.0 to 1.22.0, jaeger instance is upgrading from version 1.19.2 to 1.21.0 instead of 1.22.0.

### DISCUSSION THREAD

**Discussion by Gulnaz2021** [jaegertracing/jaeger-operator#1485] [General]
When upgrading the jaeger operator version from 1.19.0 to 1.22.0, jaeger instance is upgrading from version 1.19.2 to 1.21.0 instead of 1.22.0.

Environment: GKE Cluster

**Jaeger operator pod logs with version 1.19.0:**

time="2021-06-24T06:39:27Z" level=info msg=Versions arch=amd64 identity=do-jaeger-qa3.jaeger-operator jaeger=1.19.2 jaeger-operator=v1.19.0 operator-sdk=v0.18.2 os=linux version=go1.14.4
time="2021-06-24T06:39:41Z" level=info msg="Consider running the operator in a cluster-wide scope for extra features"
time="2021-06-24T06:39:55Z" level=info msg="Auto-detected the platform" platform=kubernetes
time="2021-06-24T06:39:55Z" level=info msg="Auto-detected ingress api" ingress-api=networking
time="2021-06-24T06:39:55Z" level=info msg="Automatically adjusted the 'es-provision' flag" es-provision=no
time="2021-06-24T06:39:55Z" level=info msg="Automatically adjusted the 'kafka-provision' flag" kafka-provision=yes
time="2021-06-24T06:39:55Z" level=info msg="The service account running this operator does not have the role 'system:auth-delegator', consider granting it for additional capabilities"
time="2021-06-24T06:40:09Z" level=warning msg="could not generate and serve custom resource metrics" error="discovering resource information failed for Jaeger in jaegertracing.io/v1: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request"
time="2021-06-24T06:40:37Z" level=warning msg="could not create ServiceMonitor object" error="unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request"
time="2021-06-24T06:40:37Z" level=debug msg="Not running on OpenShift, so won't configure OAuthProxy imagestream."
time="2021-06-24T06:45:51Z" level=debug msg="Reconciling Jaeger" execution="2021-06-24 06:45:51.750561356 +0000 UTC" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="configured this operator as the owner of the CR" execution="2021-06-24 06:45:51.750561356 +0000 UTC" instance=jaeger-instance namespace=do-jaeger-qa3 operator-identity=do-jaeger-qa3.jaeger-operator
time="2021-06-24T06:45:51Z" level=debug msg="Reconciling Jaeger" execution="2021-06-24 06:45:51.76812921 +0000 UTC" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="Strategy chosen" instance=jaeger-instance namespace=do-jaeger-qa3 strategy=allinone
time="2021-06-24T06:45:51Z" level=debug msg="Creating all-in-one deployment" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="Assembling the UI configmap" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="Assembling the Sampling configmap" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="Assembling an all-in-one deployment" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="skipping agent daemonset" instance=jaeger-instance namespace=do-jaeger-qa3 strategy=
time="2021-06-24T06:45:51Z" level=debug msg="creating service account" account=jaeger-instance instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="creating config maps" configMap=jaeger-instance-sampling-configuration instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="creating config maps" configMap=jaeger-instance-ui-configuration instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="creating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-collector-headless
time="2021-06-24T06:45:51Z" level=debug msg="creating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-collector
time="2021-06-24T06:45:51Z" level=debug msg="creating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-query
time="2021-06-24T06:45:51Z" level=debug msg="creating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-agent
time="2021-06-24T06:45:51Z" level=debug msg="creating deployment" deployment=jaeger-instance instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:51Z" level=debug msg="Deployment doesn't exist yet." name=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T06:45:52Z" level=debug msg="Waiting for deployment to stabilize" desired=1 name=jaeger-instance namespace=do-jaeger-qa3 ready=0
time="2021-06-24T06:46:00Z" level=debug msg="Deployment has stabilized" desired=1 name=jaeger-instance namespace=do-jaeger-qa3 ready=1
time="2021-06-24T06:46:00Z" level=debug msg="Reconciling Jaeger completed" execution="2021-06-24 06:45:51.76812921 +0000 UTC" instance=jaeger-instance namespace=do-jaeger-qa3

**Jaeger operator pod logs after upgrade(1.22.0):**

time="2021-06-24T07:16:25Z" level=info msg=Versions arch=amd64 identity=do-jaeger-qa3.jaeger-operator jaeger=1.21.0 jaeger-operator=v1.22.0 operator-sdk=v0.18.2 os=linux version=go1.14.15
time="2021-06-24T07:16:39Z" level=info msg="Consider running the operator in a cluster-wide scope for extra features"
time="2021-06-24T07:16:53Z" level=info msg="Auto-detected the platform" platform=kubernetes
time="2021-06-24T07:16:53Z" level=info msg="Auto-detected ingress api" ingress-api=networking
time="2021-06-24T07:16:53Z" level=info msg="Automatically adjusted the 'es-provision' flag" es-provision=no
time="2021-06-24T07:16:53Z" level=info msg="Automatically adjusted the 'kafka-provision' flag" kafka-provision=yes
time="2021-06-24T07:17:07Z" level=warning msg="could not generate and serve custom resource metrics" error="discovering resource information failed for Jaeger in jaegertracing.io/v1: unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request"
time="2021-06-24T07:17:35Z" level=warning msg="could not create ServiceMonitor object" error="unable to retrieve the complete list of server APIs: external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request"
time="2021-06-24T07:17:35Z" level=debug msg="Not running on OpenShift, so won't configure OAuthProxy imagestream."
time="2021-06-24T07:17:35Z" level=debug msg="Reconciling Jaeger" execution="2021-06-24 07:17:35.984837647 +0000 UTC" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:35Z" level=debug msg="Reconciling Deployment" name=jaeger-instance namespace=do-jaeger-qa3
E0624 07:17:35.987452       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1.Namespace: namespaces is forbidden: User "system:serviceaccount:do-jaeger-qa3:jaeger-operator" cannot list resource "namespaces" in API group "" at the cluster scope
time="2021-06-24T07:17:35Z" level=debug msg="Strategy chosen" instance=jaeger-instance namespace=do-jaeger-qa3 strategy=allinone
time="2021-06-24T07:17:35Z" level=debug msg="Creating all-in-one deployment" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:35Z" level=debug msg="Assembling the UI configmap" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:35Z" level=debug msg="Assembling the Sampling configmap" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:35Z" level=debug msg="Assembling an all-in-one deployment" instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:35Z" level=debug msg="skipping agent daemonset" instance=jaeger-instance namespace=do-jaeger-qa3 strategy=
time="2021-06-24T07:17:36Z" level=debug msg="Reconciling Deployment" name=jaeger-operator namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="updating service account" account=jaeger-instance instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="updating config maps" configMap=jaeger-instance-ui-configuration instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="updating config maps" configMap=jaeger-instance-sampling-configuration instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="updating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-collector-headless
time="2021-06-24T07:17:36Z" level=debug msg="updating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-collector
time="2021-06-24T07:17:36Z" level=debug msg="updating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-query
time="2021-06-24T07:17:36Z" level=debug msg="updating service" instance=jaeger-instance namespace=do-jaeger-qa3 service=jaeger-instance-agent
time="2021-06-24T07:17:36Z" level=debug msg="updating deployment" deployment=jaeger-instance instance=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="Deployment has stabilized" desired=1 name=jaeger-instance namespace=do-jaeger-qa3 ready=1
time="2021-06-24T07:17:36Z" level=debug msg="Reconciling Deployment" name=jaeger-instance namespace=do-jaeger-qa3
time="2021-06-24T07:17:36Z" level=debug msg="Reconciling Deployment" name=jaeger-instance namespace=do-jaeger-qa3

Can you please provide some resolution here, whether this is how it is suppose to be, that is, with Jaeger operator version 1.22.0, jaeger instance version is 1.21.0. Or there is any issue or bug?

**Comment by jpkrohling** [comment]
Could you please use 1.22.1 instead? The operator v1.22.0 was released without tracking Jaeger 1.22 (#1426)
